script/tool/test/pubspec_check_command_test.dart

```diff
@@ -1727,54 +1727,68 @@ ${_topicsSection()}
           output,
           containsAllInOrder(<Matcher>[
             contains(
-                'The following unexpected non-local dependencies were found:\n'
-                '  allow_pinned\n'
-                'Please see https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#Dependencies '
-                'for more information and next steps.'),
+                '  The following unexpected non-local dependencies were found:\n'
+                '    allow_pinned\n'
+                '  Please see https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#Dependencies\n'
+                '  for more information and next steps.'),
           ]),
         );
       });
 
-      test('fails when integration_test, flutter_test or test are used in non dev dependency',
-          () async {
-        final RepositoryPackage package =
-            createFakePackage('a_package', packagesDir, examples: <String>[]);
-
-        package.pubspecFile.writeAsStringSync('''
+      group('dev dependencies', () {
+        const List<String> packages = <String>[
+          'build_runner',
+          'integration_test',
+          'flutter_test',
+          'mockito',
+          'pigeon',
+          'test',
+        ];
+        for (final String dependency in packages) {
+          test('fails when $dependency is used in non dev dependency',
```

Comment:
I switched the structure to a loop-constructed test set where each test only checks one package, because it's easier to ensure that everything is actually being covered (e.g., that we don't add a package to the list of dependencies, but forget to add it to the error message, so that it looks like it's being tested but actually isn't).